### PR TITLE
Migrate AWSGameLift gem CDK application to CDKv2

### PR DESCRIPTION
**Pull request by lmbr-pip**
Signed-off-by: Pip Potter <61438964+lmbr-pip@users.noreply.github.com>

## What does this PR do?
Update the include AWS GameLift CDK application to AWS CDK v2. CDK v2 is the new long term stable version.

## How was this PR tested?
Ran ```cdk synth``` without issue using CDKv2 version 2.38.1.

### CODE REVIEW

File: Gems/AWSGameLift/cdk/README.md [hunk 1]
```diff
@@ -18,7 +19,11 @@ you can create the virtualenv manually.
 See https://docs.aws.amazon.com/cdk/latest/guide/getting_started.html about for information about how to set up
 the prerequisites for CDK development.
 
-To manually create a virtualenv on MacOS and Linux:
+
+>Note: This stack is for CDK v2 (the latest CDK version, if you are working with CDKv1 stacks please use the CDKv1 version of this application).
```
Comment (allisaurus) [inline Gems/AWSGameLift/cdk/README.md:23]: nit: missing space between `>` and "Note:...`

File: Gems/AWSGameLift/cdkv1/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+
+# Welcome to O3DE GameLift Sample Project!
```
Comment (allisaurus) [inline Gems/AWSGameLift/cdkv1/README.md:2]: maybe we should add a note to the top if this file about this being the old project / see `cdk` folder for newer CDKv2 version

**Review by allisaurus — COMMENTED** [review]
LGTM! Just a couple small things:

* could you amend the commit msg to say "**AWSGameLift gem** CDK app..." so it's apparent from the commit history which gem was migrated?
* a few small format/notation comments, indicated inline.

File: Gems/AWSGameLift/cdkv1/README.md (hunk 2 above)
Comment (lmbr-pip) [reply Gems/AWSGameLift/cdkv1/README.md:2]: Sure can add that.

File: Gems/AWSGameLift/cdk/README.md (hunk 1 above)
Comment (lmbr-pip) [reply Gems/AWSGameLift/cdk/README.md:23]: Will fix

**Comment by lmbr-pip** [comment]
Merging all these changes into feature branch, so closing this PR

**Pull request closed without merging**